tests/pmap_test.py

```diff
@@ -208,6 +208,34 @@ def testNestedShardingAndStacking(self, device_mesh_shape):
     self.assertEqual(ans.shape, expected.shape)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  def testPartiallyMapped(self):
+    f = pmap(lambda x, y: x - lax.psum(y, 'i'), axis_name='i', in_axes=(None, 0))
+
+    shape = (xla_bridge.device_count(), 4)
+    x = 3.
+    y = onp.arange(prod(shape), dtype=onp.float32).reshape(shape)
+    expected = onp.broadcast_to(x - onp.sum(y, 0, keepdims=True), shape)
+
+    ans = f(x, y)
+    self.assertAllClose(ans, expected, check_dtypes=False)
+
+  @parameterized.named_parameters(
+      {"testcase_name": "_mesh={}".format(device_mesh_shape).replace(" ", ""),
+       "device_mesh_shape": device_mesh_shape}
+      for device_mesh_shape in [(1, 1), (2, -1), (-1, 2)])
+  def testPartiallyMappedNested(self, device_mesh_shape):
+    mesh_shape = self._getMeshShape(device_mesh_shape)
+
+    f = pmap(lambda x, y: x - lax.psum(y, 'i'), axis_name='i', in_axes=(None, 0))
+    f = pmap(f, axis_name='j', in_axes=(None, 0))
+
+    x = 3.
+    y = onp.arange(prod(mesh_shape), dtype=onp.float32).reshape(mesh_shape)
+    expected = onp.broadcast_to(x - onp.sum(y, 1, keepdims=True), mesh_shape)
+
+    ans = f(x, y)
+    self.assertAllClose(ans, expected, check_dtypes=False)
+
```

Comment:
Consider adding a test that returns a replicated parameter (and test that the sharding spec is not replicated), just because it's kind of an interesting edge case. I'm pretty sure it works, it would more be to demonstrate the behavior, and maybe we'd want to change it someday.

Reply:
Good point, will add that! And I don't think we want to change the behavior, unless the user passes `out_axes=None`.

Reply:
Good point that we probably shouldn't change the default behavior, woo.